src/corefx/System.Globalization.Native/pal_collation.c

```diff
@@ -588,47 +592,153 @@ int32_t GlobalizationNative_IndexOfOrdinalIgnoreCase(
 }
 
 /*
- Return value is a "Win32 BOOL" (1 = true, 0 = false)
- */
-int32_t GlobalizationNative_StartsWith(
-                        SortHandle* pSortHandle,
-                        const UChar* lpTarget,
-                        int32_t cwTargetLength,
-                        const UChar* lpSource,
-                        int32_t cwSourceLength,
-                        int32_t options)
+ collation element is an int used for sorting. It consists of 3 components:
+    * primary - first 16 bits, representing the base letter
+    * secondary - next 8 bits, typically an accent
+    * tertiary - last 8 bits, typically the case
+
+An example (the numbers are made up to keep it simple)
+    a: 1 0 0
+    ą: 1 1 0
+    A: 1 0 1
+    Ą: 1 1 1
+
+    this method returns a mask that allows for characters comparison using specified Collator Strength
+*/
+static int32_t GetCollationElementMask(UColAttributeValue strength)
 {
-    int32_t result = FALSE;
-    UErrorCode err = U_ZERO_ERROR;
-    const UCollator* pColl = GetCollatorFromSortHandle(pSortHandle, options, &err);
+    switch (strength)
+    {
+        case UCOL_PRIMARY:
+            return UCOL_PRIMARYORDERMASK;
+        case UCOL_SECONDARY:
+            return UCOL_PRIMARYORDERMASK | UCOL_SECONDARYORDERMASK;
+        default:
+            return UCOL_PRIMARYORDERMASK | UCOL_SECONDARYORDERMASK | UCOL_TERTIARYORDERMASK;
```

Comment:
Theoretically we shouldn't get strength different than the primary and the secondary. You may add assert in the default case.

Reply:
I am not sure about this. When we call  `ucol_open(lpLocaleName, &err)` it returns a collator with strength == 2 (tertiary) and later we just clone it without setting the strength in explicit way

Reply:
The following code just print ```strength = UCOL_PRIMARY```. where did you see it returns tertiary

```C#
static void PrintDefaultCollator()
{
    UErrorCode err = U_ZERO_ERROR;
    UCollator* pCollator = ucol_open("en_US", &err);
    if (U_FAILURE(err))
    {
        printf("ucol_open('en_US', ...) failed with error %d\n", err);
        return;
    }

    UColAttributeValue attributeValue = ucol_getAttribute(pCollator, UCOL_STRENGTH, &err);
    if (!U_SUCCESS(err))
    {
        printf("ucol_getAttribute(...) failed with error %d\n", err);
        return;
    }

    switch (attributeValue)
    {
        case UCOL_PRIMARY: printf("strength = UCOL_PRIMARY\n"); break;
        case UCOL_SECONDARY: printf("strength = UCOL_SECONDARY\n"); break;
        case UCOL_TERTIARY: printf("strength = UCOL_PRIMARY\n"); break;
        default: printf("strength = %d\n", attributeValue);
    }
}
```

Reply:
by the way UCOL_PRIMARY = 2

Reply:
never mind. you are right.

Reply:
Maybe the assert should be on the UCOL_PRIMARY case then, because you'll always will have secondary and tertiary only. I was confused by the default strength. And you'll not need the check 

```
strength >= UCOL_SECONDARY inside SimpleStartsWith_Iterators.
```